server/src/main/java/com/paypal/selion/utils/ConfigParser.java

```diff
@@ -58,9 +66,13 @@ public static ConfigParser parse() {
      * @param file
      *            the SeLion Grid config file to use
      */
-    public static void setConfigFile(String file) {
+    public static ConfigParser setConfigFile(String file) {
         LOGGER.entering(file);
-        configFile = file;
+        if (configuration == null) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Singleton is not thread safe](https://www.codacy.com/app/SeLion/paypal-SeLion/file/1595242861/issues/source?bid=2928425&fileBranchId=3008049#l71)